pkg/controllers/gateway/suite_test.go

```diff
@@ -71,10 +85,308 @@ var _ = BeforeSuite(func() {
 	Expect(err).NotTo(HaveOccurred())
 	Expect(k8sClient).NotTo(BeNil())
 
+	k8sManager, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme: scheme.Scheme,
+	})
+	Expect(err).ToNot(HaveOccurred())
+
+	err = (&GatewayClassReconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+	}).SetupWithManager(k8sManager)
+	Expect(err).ToNot(HaveOccurred())
+
+	err = (&GatewayReconciler{
+		Client: k8sManager.GetClient(),
+		Scheme: k8sManager.GetScheme(),
+	}).SetupWithManager(k8sManager)
+	Expect(err).ToNot(HaveOccurred())
+
+	go func() {
+		defer GinkgoRecover()
+		err = k8sManager.Start(ctx)
+		Expect(err).ToNot(HaveOccurred(), "failed to run manager")
+	}()
 })
 
 var _ = AfterSuite(func() {
+	cancel()
 	By("tearing down the test environment")
 	err := testEnv.Stop()
 	Expect(err).NotTo(HaveOccurred())
 })
+
+var _ = Describe("GatewayClassController", func() {
+	Context("testing gatewayclass controller", func() {
+		var gatewayclass *gatewayv1beta1.GatewayClass
+		BeforeEach(func() {
+			gatewayclass = &gatewayv1beta1.GatewayClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "mctc-gw-istio-instance-per-cluster",
+					Namespace: "default",
+				},
+				Spec: gatewayv1beta1.GatewayClassSpec{
+					ControllerName: "kuadrant.io/mctc-gw-controller",
+				},
+			}
+		})
+
+		AfterEach(func() {
+			// Clean up GatewayClasses
+			gatewayclassList := &gatewayv1beta1.GatewayClassList{}
+			err := k8sClient.List(ctx, gatewayclassList, client.InNamespace("default"))
+			Expect(err).NotTo(HaveOccurred())
+			for _, gatewayclass := range gatewayclassList.Items {
+				err = k8sClient.Delete(ctx, &gatewayclass)
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
+
+		It("should accept a gatewayclass for this controller", func() {
+			Expect(k8sClient.Create(ctx, gatewayclass)).To(BeNil())
+			createdGatewayclass := &gatewayv1beta1.GatewayClass{}
+			gatewayclassType := types.NamespacedName{Name: gatewayclass.Name, Namespace: gatewayclass.Namespace}
+
+			// Exists
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, gatewayclassType, createdGatewayclass)
+				return err == nil
+			}, time.Second*10, time.Millisecond*250).Should(BeTrue())
```

Comment:
I imagine we might find we repeat ourselves a long with these durations. In GLBC we had something like `testtime.Short, ` etc that were just set to common durations